utilities/backupable/backupable_db.cc

```diff
@@ -1722,13 +1727,36 @@ Status BackupEngineImpl::RestoreDBFromBackup(const RestoreOptions& options,
     }
     // 3. Construct the final path
     // kWalFile lives in wal_dir and all the rest live in db_dir
-    dst = ((type == kWalFile) ? wal_dir : db_dir) + "/" + dst;
+    if (type == kWalFile) {
+      dst = wal_dir + "/" + dst;
+      if (options_.sync && !wal_dir_for_fsync) {
+        s = db_env_->NewDirectory(wal_dir, &db_dir_for_fsync);
```

Comment:
s/db_dir_for_fsync/wal_dir_for_fsync/

Reply:
Nice catch! If only there were a tool for that. https://scan.coverity.com/o/oss_success_stories/45